arm9/source/filesys/filetype.c

```diff
@@ -20,7 +20,7 @@ u64 IdentifyFileType(const char* path) {
     const u8 png_magic[] = { PNG_MAGIC };
 
     if (!path) return 0; // safety
-    u8 ALIGN(32) header[0x200]; // minimum required size
+    u8 ALIGN(32) header[0x2B8]; // minimum required size
```

Comment:
I'm not sure if having a buffer with a size that's not 32byte aligned is safe when using ALIGN(32) - if it's a minimum required, just align it to the next 32byte alignment.

Reply:
Depends on the circumstances, if one's expecting the compiler to align up it's size too, leaving space in front, there's no guarantee here, and sizeof() still does give its normal size. If just expecting compiler to just align buffer position to expected alignment, then it should safe (assuming no issues on the compiler).
In this function, there should be no concern, but I can still change it if you prefer so.

Reply:
I've stopped trusting aligned buffers on GCC since the devs seem to change their mind on what exactly to assume every one or two major versions - a recent GCC upgrade broke aligned/packed buffers and debugging it took ages, and I'd rather not have something similar again.

I do expect it to pad the buffer size to the alignment, but it'd be better to explicitly do it in the code.

Reply:
Yeah I get ya.
I'll change it then